src/metabase/driver/sql_jdbc/connection.clj

```diff
@@ -38,6 +38,19 @@
   driver/dispatch-on-initialized-driver
   :hierarchy #'driver/hierarchy)
 
+(defmulti connection-type+details->spec
+  "Given a Database `details-map` and a `connection-type`, return a JDBC connection spec. Driver authors should implement
+   this method if they need to support multiple connection types. If you only need to support a single connection type,
+   you can just implement [[connection-details->spec]] instead."
+  {:arglists '([driver connection-type details-map])}
```

Comment:
If you add new driver methods you should mention them in the https://github.com/metabase/metabase/blob/master/docs/developers-guide/driver-changelog.md dox and also include `:added` metadata